Userland/Libraries/LibAudio/QOALoader.cpp

```diff
@@ -0,0 +1,258 @@
+    case 7:
+    case 8:
+        for (size_t sample = 0; sample < header.sample_count; ++sample) {
+            target[sample] = {
+                static_cast<float>(channels[0][sample]) / static_cast<float>(NumericLimits<i16>::max()),
+                static_cast<float>(channels[1][sample]) / static_cast<float>(NumericLimits<i16>::max()),
+            };
+        }
+        break;
+    default:
+        VERIFY_NOT_REACHED();
+    }
+
+    return {};
+}
+
+LoaderSamples QOALoaderPlugin::get_more_samples(size_t max_samples_to_read_from_input)
+{
+    // Because each frame can only have so many inter-channel samples (quite a low number),
+    // we just load frames until the limit is reached, but at least one.
+    // This avoids caching samples in the QOA loader, simplifying state management.
+
+    if (max_samples_to_read_from_input < QOA::max_frame_samples)
+        return LoaderError { LoaderError::Category::Internal, LOADER_TRY(m_stream->tell()), "QOA loader is not capable of reading less than one frame of samples"sv };
+
+    ssize_t const remaining_samples = static_cast<ssize_t>(m_total_samples - m_loaded_samples);
+    if (remaining_samples <= 0)
+        return FixedArray<Sample> {};
+    size_t const samples_to_read = min(max_samples_to_read_from_input, remaining_samples);
+    auto is_first_frame = m_loaded_samples == 0 ? IsFirstFrame::Yes : IsFirstFrame::No;
+
+    auto samples = LOADER_TRY(FixedArray<Sample>::create(samples_to_read));
+    size_t current_loaded_samples = 0;
+
+    while (current_loaded_samples < samples_to_read) {
+        auto slice_to_load_into = samples.span().slice(current_loaded_samples, min(QOA::max_frame_samples, samples.size() - current_loaded_samples));
+        TRY(this->load_one_frame(slice_to_load_into, is_first_frame));
+        is_first_frame = IsFirstFrame::No;
+        VERIFY(slice_to_load_into.size() <= QOA::max_frame_samples);
+        current_loaded_samples += slice_to_load_into.size();
+        // The buffer wasn't large enough for the next frame.
+        if (slice_to_load_into.size() == 0)
+            break;
+    }
+    m_loaded_samples += current_loaded_samples;
+    auto trimmed_samples = LOADER_TRY(FixedArray<Sample>::create(samples.span().trim(current_loaded_samples)));
+
+    return trimmed_samples;
+}
+
+MaybeLoaderError QOALoaderPlugin::reset()
+{
+    LOADER_TRY(m_stream->seek(QOA::header_size, AK::SeekMode::SetPosition));
+    m_loaded_samples = 0;
+    // Read the first frame, then seek back to the beginning. This is necessary since the first frame contains the sample rate and channel count.
+    auto frame_samples = LOADER_TRY(FixedArray<Sample>::create(QOA::max_frame_samples));
+    auto span = frame_samples.span();
+    LOADER_TRY(load_one_frame(span, IsFirstFrame::Yes));
+
+    LOADER_TRY(m_stream->seek(QOA::header_size, AK::SeekMode::SetPosition));
+    m_loaded_samples = 0;
+    return {};
+}
+
+MaybeLoaderError QOALoaderPlugin::seek([[maybe_unused]] int sample_index)
```

Comment:
`sample_index` is used even if VERIFY were to be compiled out so the `[[maybe_unused]]` is no longer needed right?